src/tools/kdb/external.c

```diff
@@ -0,0 +1,205 @@
+/**
+ * @file
+ *
+ * @brief Code to support external programs
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#include <command.h>
+#include <external.h>
+#include <kdbease.h>
+
+#include <errno.h>
+#include <kdberrors.h>
+#include <spawn.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <sys/wait.h>
+#endif
+
+extern char ** environ;
+
+static void insert (StringMap * map, char * key, char * value)
+{
+	for (int i = 0; i < map->size; i++)
+	{
+		if (elektraStrCmp (map->array[i].commandName, key) == 0)
+		{
+			map->array[i].binary = value;
+			return;
+		}
+	}
+	KeyValue newPair;
+	newPair.commandName = key;
+	newPair.binary = value;
+	map->array[map->size++] = newPair;
+}
+
+char * getExternalBin (StringMap * map, const char * key)
+{
+	for (int i = 0; i < map->size; i++)
+	{
+		if (elektraStrCmp (map->array[i].commandName, key) == 0)
+		{
+			return map->array[i].binary;
+		}
+	}
+	return NULL;
+}
+
+void freeExternalBinaryMap (StringMap * map)
+{
+	for (int i = 0; i < map->size; i++)
+	{
+		if (map->array[i].commandName != NULL)
+		{
+			elektraFree (map->array[i].commandName);
+			map->array[i].commandName = NULL;
+		}
+		if (map->array[i].binary != NULL)
+		{
+			elektraFree (map->array[i].binary);
+			map->array[i].binary = NULL;
+		}
+	}
+	map->size = 0;
+}
+
+int tryLoadExternal (char * commandName, StringMap * map)
+{
+	char * execPathPtr = getenv ("KDB_EXEC_PATH");
+	bool found = false;
+	char path[1024] = { 0 };
```

Comment:
If you're not using `malloc` to support arbitrary path lengths, please use

```suggestion
	char path[PATH_MAX] = { 0 };
```

It's still possible that paths are longer than `PATH_MAX`, but it's at least better than a magic number